### DISCUSSION TITLE
Why are the Release Binaries not stripped?

### DISCUSSION THREAD

**Discussion by iranvir** [firecracker-microvm/firecracker#2751] [Q&A] [answered]
Hey, folks! I recently noticed that the binaries we get from releases page are not stripped.

```
$ file firecracker-v0.24.6-x86_64
firecracker-v0.24.6-x86_64: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), statically linked, BuildID[sha1]=284d01c5addbdee2a2eadd921700322f305c823b, **not stripped**
```
I used `llvm-strip` to strip the binary but I am concerned it might mess something or the other. Is it safe to do so? And if not can I know the reason behind it?

**Comment by raduweiss** [comment]
Hi @iranvir . The team is looking a few other items right now, and we'll get back to you in a few days.

**Reply by iranvir** [reply]
Thank you!

**Comment by raduiliescu** [comment, marked as answer]
Hi @iranvir!

Please see this past issues and PRs on strip:
https://github.com/firecracker-microvm/firecracker/issues/1652
https://github.com/firecracker-microvm/firecracker/pull/1957
https://github.com/firecracker-microvm/firecracker/pull/2567

We have decided to strip only the debug symbols from our releases, since we want to have some information in the stack traces or when debugging. 
I've you want to strip the further, there shouldn't be any problem. Please note stripping will have effect only on the disk space, will not affect the memory since the symbols are in a different section that is loaded in the memory only when needed.
